tests/general/views/test_consolidated_profile_view.py

```diff
@@ -0,0 +1,566 @@
+            {"gender": "female", "age": "16", "count": 3},
+        ]
+        indicator = self.create_versioned_data(
+            version2, geography, profile, data=data
+        )
+        self.create_highlight(
+            profile, indicator, "female", "highlight_v2"
+        )
+
+        # Request without any version
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        assert response.data["profile"]["highlights"] == [
+            {'label': 'highlight_v1', 'value': 2.0, 'method': 'absolute_value'}
+        ]
+
+        # Request with version2
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'version': version2.name,
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        assert response.data["profile"]["highlights"] == [
+            {'label': 'highlight_v2', 'value': 3.0, 'method': 'absolute_value'}
+        ]
+
+    def test_highlight_sibling_with_multiversion(self):
+        version2 = VersionFactory()
+        geo3 = self.root.add_child(code="GT")
+        self.create_boundary(geo3, version2)
+        self.create_boundary(self.geo1, version2)
+        configuration = self.hierarchy.configuration
+        configuration["versions"] = [self.version.name, version2.name]
+        self.hierarchy.configuration = configuration
+        self.hierarchy.save()
+
+        # self.version data
+        data = {
+            "geo1": [{"gender": "female", "age": "16", "count": 2}],
+            "geo2": [{"gender": "female", "age": "16", "count": 4}],
+            "geo3": [{"gender": "female", "age": "16", "count": 6}],
+        }
+
+        dataset1 = self.create_dataset(self.profile, self.version)
+        dataset2 = self.create_dataset(self.profile, version2)
+        for dataset in [dataset1, dataset2]:
+            self.multiple_datasetdata(dataset, self.geo1, data["geo1"])
+            self.multiple_datasetdata(dataset, self.geo2, data["geo2"])
+            self.multiple_datasetdata(dataset, geo3, data["geo3"])
+            indicator = self.create_indicator(dataset, "gender")
+            self.create_highlight(
+                self.profile, indicator, "female", "highlight_v1", "sibling"
+            )
+
+        response = self.get(
+            'all-details',
+            profile_id=self.profile.pk,
+            geography_code=self.geo1.code,
+            data={
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        assert response.data["profile"]["highlights"] == [
+            {'label': 'highlight_v1', 'value': 0.3333333333333333, 'method': 'sibling'}
+        ]
```

Comment:
I don't understand.

1. geo 1, 2, and 3 are siblings
2. 2+4+6=12
3. 2/12=0.166666666

I think at the top of this method you need to lay out the structure of hierarchy - I can't figure it out in my head.

Reply:
removing loop 👍